sdks/python/apache_beam/runners/portability/fn_api_runner_transforms.py

```diff
@@ -788,14 +799,34 @@ def make_stage(base_stage, transform_id, extra_must_follow=()):
             inputs=dict(transform.inputs, **{main_input_tag: paired_pcoll_id}),
             outputs={'out': split_pcoll_id})
 
+        if common_urns.composites.RESHUFFLE.urn in context.known_runner_urns:
```

Comment:
Is the plan to eventually retrieve the known URNs dynamically?

Reply:
Either that or make RESHUFFLE a required primitive. (Or inject a full composite reshuffle here, but that has a pile of ufuncs in it.)